onedal/covariance/incremental_covariance.py

```diff
@@ -121,21 +161,24 @@ def finalize_fit(self, queue=None):
         self : object
             Returns the instance itself.
         """
-        params = self._get_onedal_params(self._dtype)
-        result = self._get_backend(
-            "covariance",
-            None,
-            "finalize_compute",
-            self._policy,
-            params,
-            self._partial_result,
-        )
-        if daal_check_version((2024, "P", 1)) or (not self.bias):
-            self.covariance_ = from_table(result.cov_matrix)
-        else:
-            n_rows = self._partial_result.partial_n_rows
-            self.covariance_ = from_table(result.cov_matrix) * (n_rows - 1) / n_rows
-
-        self.location_ = from_table(result.means).ravel()
+        if self._need_to_finalize:
```

Comment:
This comment is out of the scope of the current PR.
I think make sense to do refactoring in follow up. It seems to me that better to have some kind of common BaseIncremenatalEstimator class with such attribute documented well and base abstract methods stored there